### PR TITLE
docs: update bad undo-add links

### PR DESCRIPTION
**Pull request by jorgeorpinel**
per https://github.com/iterative/dvc.org/runs/1179565776

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
p.s. should I also add a temporary redirect Ivan?

**Comment by jorgeorpinel** [comment]
p.p.s. hey @jeremydesroches, how would we inspect how much traffic `dvc.org/doc/user-guide/updating-tracked-files`
 used to get? The Search Console says "URL is not on Google": https://search.google.com/u/1/search-console/inspect?resource_id=sc-domain%3Advc.org&id=1BHK2EaGld46LRxtSJ8TlA

**Comment by shcheklein** [comment]
> p.s. should I also add a temporary redirect Ivan?

nah, I think we are fine here

CI checks are still broken ^^

**Review by shcheklein — CHANGES_REQUESTED** [review]
CI is broken

File: content/docs/command-reference/remove.md [hunk 1]
```diff
@@ -34,8 +34,8 @@ Note that in the case of `.dvc` file `targets`, the tracked files or directories
 `dvc gc`). However, remember to run `dvc push` to save the files you actually
 want to use or share in the future.
 
-Refer to [Updating Tracked Files](/doc/user-guide/updating-tracked-files) to see
-how it can be used to replace or modify files that are tracked by DVC.
+Refer to [Updating Tracked Files](/doc/user-guide/how-to/updated-tracked-files)
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remove.md:37]:
```suggestion
Refer to [Updating Tracked Files](/doc/user-guide/how-to/update-tracked-files)
```

File: content/docs/command-reference/unprotect.md [hunk 2]
```diff
@@ -26,8 +26,8 @@ modifying file links.)
 Running `dvc unprotect` guarantees that the target files or directories
 (`targets`) in the workspace are physically "unlinked" from the cache and can be
 safely updated. Read the
-[Update a Tracked File](/doc/user-guide/updating-tracked-files) guide to learn
-more on this process.
+[Update a Tracked File](/doc/user-guide/how-to/updated-tracked-files) guide to
```
Comment (jorgeorpinel) [inline content/docs/command-reference/unprotect.md:29]:
```suggestion
[Update a Tracked File](/doc/user-guide/how-to/update-tracked-files) guide to
```

File: content/docs/user-guide/large-dataset-optimization.md [hunk 3]
```diff
@@ -100,8 +100,8 @@ efficiency:
 
 > DVC avoids `symlink` and `hardlink` types by default to protect user from
 > accidental cache corruption. Refer to the
-> [Update a Tracked File](/doc/user-guide/updating-tracked-files) guide to learn
-> more.
+> [Update a Tracked File](/doc/user-guide/how-to/updated-tracked-files) guide to
```
Comment (jorgeorpinel) [inline content/docs/user-guide/large-dataset-optimization.md:103]:
```suggestion
> [Update a Tracked File](/doc/user-guide/how-to/update-tracked-files) guide to
```

File: content/docs/user-guide/large-dataset-optimization.md [hunk 4]
```diff
@@ -117,9 +117,9 @@ $ dvc config cache.type hardlink,symlink
 > Refer to `dvc config cache` for more details.
 
 Note that with this `cache.type`, your workspace files will be in read-only mode
-in order to protect the cache from corruption. Please refer to the
-[Update a Tracked File](/doc/user-guide/updating-tracked-files) on how to manage
-tracked files under these cache configurations.
+in order to protect the cache from corruption. Please refer to
+[Update a Tracked File](/doc/user-guide/how-to/updated-tracked-files) on how to
```
Comment (jorgeorpinel) [inline content/docs/user-guide/large-dataset-optimization.md:121]:
```suggestion
[Update a Tracked File](/doc/user-guide/how-to/update-tracked-files) on how to
```

**Comment by jorgeorpinel** [comment]
Sorry my bad... Typo 🤦 

BTW these new checks against the deployment app are great! Thanks @shcheklein @rogermparent

**Review by shcheklein — APPROVED** [review]

**Comment by jeremydesroches** [comment]
> p.p.s. hey @jeremydesroches, how would we inspect how much traffic `dvc.org/doc/user-guide/updating-tracked-files`
> used to get? The Search Console says "URL is not on Google": https://search.google.com/u/1/search-console/inspect?resource_id=sc-domain%3Advc.org&id=1BHK2EaGld46LRxtSJ8TlA

The trick is to go into performance and then filter by the url. [This link](https://search.google.com/search-console/performance/search-analytics?resource_id=sc-domain%3Advc.org&page=*dvc.org%2Fdoc%2Fuser-guide%2Fupdating-tracked-files&num_of_months=12) should get you there on Search Console.

**Comment by jorgeorpinel** [comment]
I see. Yeah I discovered that recently on Search Console actually. But this tells me how many impressions and clicks it gets, as well as the search queries, which is useful (screenshot below). But I was wondering even more general than that: the net traffic it gets, so should probably check in Google Analytics instead?

image

**Comment by jeremydesroches** [comment]
> the net traffic it gets, so should probably check in Google Analytics instead?

Yes. Usually the clicks will line up pretty well with sessions (for organic), but [try this view in Analytics](https://analytics.google.com/analytics/web/?authuser=0#/report/content-pages/a120072346w177588519p176254157/_u.date00=20190901&_u.date01=20201002&explorer-table.plotKeys=%5B%5D&_r.drilldown=analytics.pagePath:dvc.org~2Fdoc~2Fuser-guide~2Fupdating-tracked-files&_.useg=builtin5/).

**Comment by jorgeorpinel** [comment]
Oh cool, thanks @jeremydesroches. Do you know if there's a way to add more than one page? I'd like to overlay that one with the new URL, dvc.org/doc/user-guide/how-to/update-tracked-files to see how smooth the transition was (there's a redirect form the previous one to this one).

**Comment by jeremydesroches** [comment]
> Oh cool, thanks @jeremydesroches. Do you know if there's a way to add more than one page? I'd like to overlay that one with the new URL, dvc.org/doc/user-guide/how-to/update-tracked-files to see how smooth the transition was (there's a redirect form the previous one to this one).

Yep, try this [report in Analytics](https://analytics.google.com/analytics/web/?authuser=0#/report/content-pages/a120072346w177588519p176254157/_u.date00=20200801&_u.date01=20201006&_.useg=builtin5&explorer-table.filter=tracked-files&explorer-table.plotKeys=%5B%5B%22dvc.org~2Fdoc~2Fuser-guide~2Fhow-to~2Fupdate-tracked-files%22%5D,%5B%22dvc.org~2Fdoc~2Fuser-guide~2Fupdating-tracked-files%22%5D%5D/). You can also use [compare in Search Console](https://search.google.com/search-console/performance/search-analytics?resource_id=sc-domain%3Advc.org&page=!https%3A%2F%2Fdvc.org%2Fdoc%2Fuser-guide%2Fupdating-tracked-files&compare_page=https%3A%2F%2Fdvc.org%2Fdoc%2Fuser-guide%2Fhow-to%2Fupdate-tracked-files&num_of_months=3). 

I tried out the [original link](dvc.org/doc/user-guide/updating-tracked-files) and there doesn't seem to be a working redirect at the moment. It's good that the intent is to have a redirect, and I assume that will be fixed. This is actually a pretty important thing to handle for SEO purposes, so I'll just spell out what should happen in cases like this:

1. Publish the updated page at new url: `dvc.org/original-url` becomes `dvc.org/new-url`.
2. Create 301 redirect from `dvc.org/original-url` to `dvc.org/new-url`.
3. In Search Console, request indexing for `dvc.org/new-url`.
4. In Search Console, request indexing for `dvc.org/original-url` (so it picks up the redirect immediately).
5. Provided that new content is similar or better, organic traffic and ranking should transition seamlessly.

Hope this helps!

**Comment by jeremydesroches** [comment]
> Yep, try this [report in Analytics](https://analytics.google.com/analytics/web/?authuser=0#/report/content-pages/a120072346w177588519p176254157/_u.date00=20200801&_u.date01=20201006&_.useg=builtin5&explorer-table.filter=tracked-files&explorer-table.plotKeys=%5B%5B%22dvc.org~2Fdoc~2Fuser-guide~2Fhow-to~2Fupdate-tracked-files%22%5D,%5B%22dvc.org~2Fdoc~2Fuser-guide~2Fupdating-tracked-files%22%5D%5D/). 

By the way, to produce that report: Find or search for the urls you want in the table, then check them off and select "Plot Rows".

**Comment by jorgeorpinel** [comment]
Sweet! Looks like that search box takes regexps too, which is useful to find typos in links out there. I just fixed 2 old links that now cause 404s in old GitHub comments based on this:

image

> I like that GA let's you look at weekly graphs too, which are much more smooth. If you find a way to do that in SC please lmk.

Page Compare on SC is also good but you can only have 2.

**Comment by jorgeorpinel** [comment]
> This is actually a pretty important thing to handle for SEO purposes, so I'll just spell out what should happen

Actually we completely flopped on this this time. Not a big deal since we're talking about 100 visits/week but totally agree with those steps. We usually setup the redirects (which get removed later at some point) but we never reindex on SC — good tip, thanks.

I would add a step: to find any old links we have control of and update them, so the redirect can be removed later on.  We already do this in all the docs of course, since it's easy to grep the old URL, but they're also common in old GH comments, for example (maybe some 3rd party blogs we wrote?) which tend to not get updated.

**Comment by jeremydesroches** [comment]
> Sweet! Looks like that search box takes regexps too, which is useful to find typos in links out there. I just fixed 2 old links that now cause 404s in old GitHub comments...

Nice @jorgeorpinel! Yes, you can do some amazing things with regexp in GA. Another spot I've used them is with Filters... easy to get carried away, but if you ever find visits coming in with strange event/channel/tag data, a regexp filter can clean everything up nicely. Only for data going forward though.

>  I like that GA let's you look at weekly graphs too, which are much more smooth. If you find a way to do that in SC please lmk.

> Page Compare on SC is also good but you can only have 2.

Yes, SC is pretty limited... BUT you can view SC data in GA if you [enable the connection](https://analytics.google.com/analytics/web/?authuser=0#/report/acquisition-sc-landingpages/a120072346w177588519p176254157/_u.date00=20200907&_u.date01=20201006&_.goalOption=ALL&_.useg=builtin1/)! Maybe you can get the best of both... 🙂

**Comment by jeremydesroches** [comment]
> Actually we completely flopped on this this time. Not a big deal since we're talking about 100 visits/week but totally agree with those steps. We usually setup the redirects (which get removed later at some point) but we never reindex on SC — good tip, thanks.

Probably no harm done, as long as it's redirected relatively soon. The sitemap is likely being read daily, so new URLs will get picked up easily (e.g., a new blog post or doc will get indexed almost immediately). When URLs change it's a bit different: the 301 redirect assigns the previous URL's ranking to the new URL so takes its place in the index. Without a redirect the new URL _might_ be recognized as the old page, but could actually "start over". The old URL can hang around in the index for quite a while, too.

> I would add a step: to find any old links we have control of and update them, so the redirect can be removed later on. 

Yes, a very practice if the scope of your content allows for it. It's definitely better to have a well-maintained redirects file than hundreds of historical entries — but it happens. 😒

**Comment by jorgeorpinel** [comment]
Thanks for the tips about Filters and on linking SC and GA @jeremydesroches, will check it out.

> Probably no harm done, as long as it's redirected relatively soon

Well, the change happened weeks ago already and as you may have noticed in the merged PR comment where I mentioned you, there was some damage done 😓 but if you want (and it can be good so you learn this) feel free to setup the redirect in [redirects-list.json](https://github.com/iterative/dvc.org/blob/master/redirects-list.json) with a small PR.

**Pull request merged**